src/UI/SCMPane.re

```diff
@@ -0,0 +1,118 @@
+open Oni_Core;
+open Oni_Model;
+open Revery;
+open Revery.UI;
+open Utility;
+
+module Styles = {
+  open Style;
+
+  let container = [padding(10), flexGrow(1)];
+
+  let text = (~theme: Theme.t, ~font: UiFont.t) => [
+    fontSize(font.fontSize),
+    fontFamily(font.fontFile),
+    color(theme.sideBarForeground),
+    textWrap(TextWrapping.NoWrap),
+    textOverflow(`Ellipsis),
+  ];
+
+  let group = [];
+
+  let groupLabel = [paddingVertical(3)];
+
+  let groupLabelText = (~theme: Theme.t, ~font: UiFont.t) => [
+    fontSize(font.fontSize *. 0.85),
+    fontFamily(font.fontFileBold),
+    color(theme.sideBarForeground),
+    textWrap(TextWrapping.NoWrap),
+    textOverflow(`Ellipsis),
+  ];
+
+  let groupItems = [marginLeft(6)];
+
+  let item = (~isHovered, ~theme: Theme.t) => [
+    isHovered
+      ? backgroundColor(theme.listHoverBackground)
+      : backgroundColor(theme.sideBarBackground),
+    paddingVertical(2),
+    cursor(MouseCursors.pointer),
+  ];
+};
+
+let%component itemView =
+              (
+                ~provider: SCM.Provider.t,
+                ~resource: SCMResource.t,
+                ~theme,
+                ~font,
+                ~workspace: Workspace.t,
+                (),
+              ) => {
+  open Base;
+  let%hook (isHovered, setHovered) = Hooks.state(false);
+  let onMouseOver = _ => setHovered(_ => true);
+  let onMouseOut = _ => setHovered(_ => false);
+
+  let base =
+    Option.first_some(
+      Option.map(provider.rootUri, ~f=Uri.toFileSystemPath),
+      Option.map(workspace, ~f=w => w.workingDirectory),
+    )
+    |> Option.value(~default="/");
```

Comment:
Tried out Base a bit here. The labelled argument convention is nice, as it allowed putting the main argument first, which helps the type inference so the function argument doesn't have to be annotated. Otherwise I found the available functions a bit lacking compared to containers. `first_some` is not as nice as `choice` and `or_` in containers for example.

Reply:
> The labelled argument convention is nice, as it allowed putting the main argument first, which helps the type inference so the function argument doesn't have to be annotated.

Oh, interesting. That makes sense.

>  Otherwise I found the available functions a bit lacking compared to containers. first_some is not as nice as choice and or_ in containers for example.

That's too bad. I like the ergonomics of `or_` and `choose` as well. I guess we could either switch to containers, or still have some `OptionEx` wraps around the `base` methods. Thanks for taking it for a test run.